File: AppMovies/app/src/main/java/com/svmdev/appmovies/data/webservice/RetrofitConfig.java
package com.svmdev.appmovies.data.webservice;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitConfig {

    private static final long DEFAULT_TIMEOUT_SECONDS = 15L;
    private final Retrofit retrofit;

    public RetrofitConfig() {
        String BASE_URL = "http://www.google.com.br";
        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BODY);
        OkHttpClient.Builder httpClient = new OkHttpClient.Builder();

        httpClient.addInterceptor(logging)
                .readTimeout(DEFAULT_TIMEOUT_SECONDS, TimeUnit.SECONDS)
                .connectTimeout(DEFAULT_TIMEOUT_SECONDS,TimeUnit.SECONDS)
                .writeTimeout(DEFAULT_TIMEOUT_SECONDS,TimeUnit.SECONDS);

        this.retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .client(httpClient.build())
                .build();

    }

    public API getAPI(){
        return this.retrofit.create(API.class);
    }

}

File: app/src/main/java/com/movies/appmoviescit/services/RetrofitConfig.kt
package com.movies.appmoviescit.services

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitConfig {

    private var client: OkHttpClient
    private var retrofit: Retrofit

    init {
        val loggingInterceptor = HttpLoggingInterceptor()
        loggingInterceptor.level = HttpLoggingInterceptor.Level.BODY

        client = OkHttpClient.Builder()
            .addInterceptor(loggingInterceptor)
            .build()

        retrofit = Retrofit.Builder()
            .client(client)
            .baseUrl(getBaseURL())
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }

    fun movieService() = retrofit.create(IMovieService::class.java)

    protected open fun getBaseURL() = MoviesConstants.API_URL_BASE

}